packages/protocol-kit/src/managers/contractManager.ts

```diff
@@ -37,30 +37,40 @@ class ContractManager {
     let safeVersion: SafeVersion
 
     if (isSafeConfigWithPredictedSafe(config)) {
-      safeVersion = config.predictedSafe.safeDeploymentConfig?.safeVersion ?? DEFAULT_SAFE_VERSION
+      safeVersion = predictedSafe?.safeDeploymentConfig?.safeVersion ?? DEFAULT_SAFE_VERSION
     } else {
-      const temporarySafeContract = await getSafeContract({
+      // We use the lastest version of the Safe contract to get the correct version of this Safe
+      const latestVersionOfTheSafeContract = await getSafeContract({
```

Comment:
I was thinking that this variable name may not be accurate, as the DEFAULT_SAFE_VERSION may not be the latest.

I would rename it to `defaultSafeContractInstance`